### DISCUSSION TITLE
Cannot add lines for unavailable for purchase variants.

### DISCUSSION THREAD

**Discussion by slamer59** [saleor/saleor#10391] [Q&A] [answered]
Hello,
I don't understand why this error is raised (in validate_variants_available_for_purchase saleor-core)
_Cannot add lines for unavailable for purchase variants._

I have check in the dashboard and I see them 
![image](https://user-images.githubusercontent.com/25283272/184861132-99c36184-8395-4e3c-9ec5-4ffda7d41167.png)

I also check the demo and I found that the request is not the same ?
![image](https://user-images.githubusercontent.com/25283272/184860320-930a0b73-76e3-4444-bec0-6b7bc7e9656c.png)

Any idea what I have missed ?

**Comment by karolkielecki** [comment]
Hi @slamer59, could you please tell me your Saleor version and other details that might be useful to reproduce the issue?

**Reply by slamer59** [reply]
Sorry for the delayed response :)

**Comment by korycins** [comment, marked as answer]
Hey @slamer59 
Can you confirm that your product is published and available for purchase?
Dashboard->products-><your-product>-> Availability section
![image](https://user-images.githubusercontent.com/23282005/185342083-0e189f1e-fd1f-44a3-866d-4cc9726d308d.png)

**Comment by slamer59** [comment]
@korycins Thanks that did the trick ! 
I didn't know where it cames from ! 

By the way, I want to know if it is possible to control the language of this text. It seems to come from core. 

Now I have this error but I will find the options 
Could not add items XXXXX Only 0 remaining in stock.

I have two warehouses but 0 allocated
![image](https://user-images.githubusercontent.com/25283272/185410056-7b577646-1b57-417b-8a56-e484522b7c9c.png)

**Reply by korycins** [reply]
> By the way, I want to know if it is possible to control the language of this text. It seems to come from core.

What kind of the text? Do you mean the `product`'s text? You can add translation of the products  `dashboard->transalations-> products`.

If you mean `Cannot add lines for unavailable for purchase variants.` that is returned from the core, then you should use `code` field returned by API. You can use `code` field to build the error message that you will display to the user.

>Now I have this error but I will find the options
Could not add items XXXXX Only 0 remaining in stock.

Make sure that you have a proper shipping zone (Dashboard->Shipping methods). Make sure that the shipping country is declared in the shipping zone, and make sure that the shipping zone is also attached to the proper channel and warehouse.


![image](https://user-images.githubusercontent.com/23282005/185414316-6cb1670c-bc5b-4372-b96b-c0185dc01bdb.png)
